tests/bluetooth/tester/src/btp.c

```diff
@@ -90,29 +87,33 @@ static void cmd_handler(void *p1, void *p2, void *p3)
 	while (1) {
 		const struct btp_handler *btp;
 		struct btp_buf *cmd;
+		struct btp_hdr *hdr;
 		uint8_t status;
 		uint16_t rsp_len = 0;
-		uint16_t len;
 
 		cmd = k_fifo_get(&cmds_queue, K_FOREVER);
+		hdr = (struct btp_hdr *)cmd->data;
 
-		LOG_DBG("cmd service 0x%02x opcode 0x%02x index 0x%02x", cmd->hdr.service,
-			cmd->hdr.opcode, cmd->hdr.index);
+		LOG_DBG("cmd service 0x%02x opcode 0x%02x index 0x%02x",
+			hdr->service, hdr->opcode, hdr->index);
 
-		len = sys_le16_to_cpu(cmd->hdr.len);
-
-		btp = find_btp_handler(cmd->hdr.service, cmd->hdr.opcode);
+		btp = find_btp_handler(hdr->service, hdr->opcode);
 		if (btp) {
-			if (btp->index != cmd->hdr.index) {
+			uint16_t len = sys_le16_to_cpu(hdr->len);
+
+			if (len > BTP_DATA_MAX_SIZE) {
+				status = BTP_STATUS_FAILED;
+			} else if (btp->index != hdr->index) {
```

Comment:
Hmm, now I wonder if it's really need as we do have `} else if ((btp->expect_len >= 0) && (btp->expect_len != len)) {` which I guess would already catch it?

Reply:
expect_len is used only for fixed size commands,  so it could stil pass it if command is variable length

Reply:
Good point :) Could introduce something like a `btp->max_len` as well, but I guess this is more generic

Reply:
yeah, we can do something more generic than BTP_HANDLER_LENGTH_VARIABLE, so that handlers for variable length commands would only have to validate variable part of command